src/marks/auto.js

```diff
@@ -211,23 +164,78 @@ export function auto(data, options) {
   if (transform) {
     if (transform === bin || transform === binX) markOptions.x = {value: X, ...xOptions};
     if (transform === bin || transform === binY) markOptions.y = {value: Y, ...yOptions};
-    markOptions = transform(transformOptions, markOptions);
   }
 
   // If zero-ness is not specified, default based on whether the resolved mark
-  // type will include a zero baseline. TODO Move this to autoSpec.
+  // type will include a zero baseline.
   if (xZero === undefined)
-    xZero = X && transform !== binX && (mark === barX || mark === areaX || mark === rectX || mark === ruleY);
+    xZero =
+      X &&
+      transform !== bin &&
+      transform !== binX &&
+      (markImpl === barX || markImpl === areaX || markImpl === rectX || markImpl === ruleY);
```

Comment:
I added the `transform !== bin` check for both xZero and yZero, which I think achieves the same thing as checking `!colorReduce && !sizeReduce` in the earlier implementation, i.e. it fixes the heatmap case.